pkg/network/network_test.go

```diff
@@ -0,0 +1,128 @@
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Testing network", func() {
+	// There is no functionality in this function yet
+	Describe("Canonicalize", func() {
+		Context("when given an empty config", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			It("should pass", func() {
+				Canonicalize(conf)
+			})
+		})
+	})
+
+	Describe("Validate", func() {
+		Context("when given a valid config", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			openshiftNetworkConf := &osv1.Network{}
+			It("should pass", func() {
+				err := Validate(conf, openshiftNetworkConf)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("when given invalid config", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{
+					RangeStart: "foo",
+				},
+				ImagePullPolicy: v1.PullPolicy("bar"),
+			}
+			openshiftNetworkConf := &osv1.Network{}
+			It("should return a compilation of errors", func() {
+				err := Validate(conf, openshiftNetworkConf)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("invalid configuration"))
+				Expect(err.Error()).To(ContainSubstring("both or none of the KubeMacPool ranges needs to be configured"))
+				Expect(err.Error()).To(ContainSubstring("requested imagePullPolicy 'bar' is not valid"))
+			})
+		})
+	})
+
+	Describe("FillDefaults", func() {
+		Context("when given valid configuration", func() {
+			newConf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			prevConfig := &opv1alpha1.NetworkAddonsConfigSpec{}
+
+			It("should successfully pass", func() {
+				err := FillDefaults(newConf, prevConfig)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+	})
+
+	Describe("IsChangeSafe", func() {
+		Context("when current and new configuration is compatible", func() {
+			newConf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			prevConfig := &opv1alpha1.NetworkAddonsConfigSpec{}
+
+			It("should pass the check", func() {
+				err := IsChangeSafe(prevConfig, newConf)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("when current and new configuration is not compatible", func() {
+			newConf := &opv1alpha1.NetworkAddonsConfigSpec{ImagePullPolicy: v1.PullAlways}
+			prevConfig := &opv1alpha1.NetworkAddonsConfigSpec{ImagePullPolicy: v1.PullIfNotPresent}
+
+			It("should fail the check", func() {
+				err := IsChangeSafe(prevConfig, newConf)
+				Expect(err).To(HaveOccurred())
+			})
+		})
+	})
+
+	Describe("Render", func() {
+		Context("when given valid arguments", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{ImagePullPolicy: v1.PullAlways, Multus: &opv1alpha1.Multus{}, LinuxBridge: &opv1alpha1.LinuxBridge{}}
+			manifestDir := "../../data"
+			openshiftNetworkConf := &osv1.Network{}
+			clusterInfo := &ClusterInfo{SCCAvailable: true, OpenShift4: false}
+
+			It("should successfully render a set of objects", func() {
```

Comment:
perhaps you could add a negative case here, this seems like an important case to cover.

Reply:
Done for invalid manifestsDir